fairscale/nn/data_parallel/sharded_ddp.py

```diff
@@ -113,6 +118,24 @@ def __init__(
         # - keep track of the grads which have already been reduced
         self._reduced_grads: Dict[OSS, int] = {}
         self._reduced_grads_max = {o: len(o.param_to_rank.values()) for o in self.sharded_optimizers}
+
+        # - setup buckets and tensor views
+        model_size = sum([p.numel() for p in self.module.parameters()])
+        if dist.get_world_size(self.process_group) <= 8:
```

Comment:
8 -> torch.cuda.device_count()?

Reply:
in that case it was about the number of ranks, but I'm not sure that it's that good of a call actually, could also be removed and be kept entirely in the user scope.

Reply:
removed, I think it would need more testing, better to remove a branch I suppose